pkg/master/tunneler/ssh.go

```diff
@@ -157,9 +157,14 @@ func (c *SSHTunneler) SecondsSinceSync() int64 {
 }
 
 func (c *SSHTunneler) SecondsSinceSSHKeySync() int64 {
-	now := c.clock.Now().Unix()
-	then := atomic.LoadInt64(&c.lastSSHKeySync)
-	return now - then
+	// If the cloud-provider doesn't support installing SSH keys this function will
+	// eventually cause the health check to fail because the time is never updated.
```

Comment:
The ssh-user option is long-deprecated, and the fact that health checks fail on it currently mean we know successful usage is limited. I am not in favor of enabling expanded use of a deprecated option. See https://groups.google.com/d/msg/kubernetes-sig-api-machinery/bwJJ93qA99o/pRTVFaIlBAAJ